core/quorum_validation.go

```diff
@@ -175,6 +177,11 @@ func (c *Core) validateTokenOwnership(cr *ConensusRequest, sc *contract.Contract
 				c.log.Error("failed to sync parent token chain", "token", pt)
 				return false, err
 			}
+			_, err = c.w.Pin(pt, wallet.WholeTokenSplitLockRole, quorumDID, cr.TransactionID, address, receiverAddress, ti[i].TokenValue)
```

Comment:
how are we identifying if the parent token is a whole token or part token , when giving the role WholeTokenSplitLockRole

Reply:
So in this case we are not identifying whether the parent token is a whole token or part token. We are just pinning the parent token from the quorum. So may be I can change the role to quorum role instead of wholeTokenSplitLockRole ?

Reply:
Maybe create a Role like "ParentTokenPinning" this would be a bit suitable for the operation we are doing